test/system/trader.js

```diff
@@ -0,0 +1,179 @@
+const cryptoClient = require('../../lib/cryptonite-client');
+const logInitialDetails = require('./initilaLogger');
+
+class Trader {
+
+  constructor(config) {
+    this.config = config;
+    this.client = new cryptoClient(this.config);
+    this.market = 'LTC/BTC';
+    this.init_state(this.config);
+  }
+
+  init_state() {
+    this._myorders = {updated: undefined, data: undefined, errors: []};
+    this.client.connect()
+      .then(() => {
+        this.client.getBalances()
+          .then(data => {
+            const balanceData = data.balances;
+            this.config.balance.assets = balanceData.btc.availableBalance;
+            this.config.balance.capital = balanceData.ltc.availableBalance;
+            this.config.state.start.assets = balanceData.btc.availableBalance;
+            this.config.state.start.capital = balanceData.ltc.availableBalance;
+            this.logInitialDetails(this.config);
+            return this.client.getMyOrders(1, 4);
+          })
+          .then((data) => {
+            this._myorders.updated = new Date();
+            this._myorders.data = data.orders;
+            console.log(this._myorders);
+            this.start_trading();
+          });
+      });
+  }
+
+  logInitialDetails(config) {
+    logInitialDetails({
+      config,
+    });
+  }
+
+  start_trading() {
+    this._execute_trading_strategy();
+  }
+
+  getBalance() {
+    return {
+      assets: this.config.balance.assets,
+      capital: this.config.balance.capital,
+      start: {
+        assets: this.config.state.start.assets,
+        capital: this.config.state.start.capital,
+        price: this.config.state.start.price,
+      }
+    };
+  }
+
+  updateBalance(type, num) {
+    if (type === 'CAPITAL') {
+      this.config.balance.capital = num;
+    } else if (type === 'ASSETS') {
+      this.config.balance.assets = num;
+    }
+  }
+
+  cancelAllOrders() {
+    let self  = this;
+    setTimeout(function() {
```

Comment:
Use of arrow functions will make code like this easier...
```
setTimeout(() => {
  this.client.cancelMarketOrders(this.market)
  // ...
}, 100);
```
i.e. no need for self = this everywhere